### PR TITLE
Planable test second try

### PR DESCRIPTION
**Pull request by VictorvManea**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # QA Task
 
+https://www.notion.so/Bug-report-8aabced30df243008d3928069ac96893?pvs=4  -  link to the bug report. Let me know if you are not able to access it, as it is the first time I am sharing a notion file, not sure how it works:)
```
Comment (tommy-anderson) [inline README.md:3]: Hey, can you please add alexursu@planable.io to the shared users.

File: cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { defineConfig } from "cypress";
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
+      // implement node event listeners here
+    },
+    baseUrl: 'http://localhost:5173',
```
Comment (tommy-anderson) [inline cypress.config.ts:8]: good use of baseUrl

File: cypress/e2e/bugs.cy.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { should } from "chai"
```
Comment (tommy-anderson) [inline cypress/e2e/bugs.cy.ts:1]: you don't need "chai" imported in all files

File: README.md (hunk 1 above)
Comment (VictorvManea) [reply README.md:3]: Done!

File: cypress/e2e/bugs.cy.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+describe('Adding items test', () => {
+  beforeEach(() => {
+    cy.visit("/").get('.text-white').should('exist');
```
Comment (tommy-anderson) [inline cypress/e2e/bugs.cy.ts:5]: There is no need to chain a get after the visit, as we don't need the yielded window object after the page finishes loading. Furthermore it's [not safe](https://docs.cypress.io/api/commands/visit#Yields) to chain after visit

File: cypress/e2e/bugs.cy.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  })
+
+  it('should not let user add empty item', () => {
+    cy.get('.w-full').type("")
```
Comment (tommy-anderson) [inline cypress/e2e/bugs.cy.ts:9]: a better approach would be to assert that the field is currently empty and press Enter without typing `""`

File: cypress/e2e/bugs.cy.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    cy.get('.hover\\:bg-blue-700').click()
+    cy.get('.cursor-pointer').should('have.text', 'Berries')
+  })
+
+  it('should not allow user to save item with negative integer as quantity', () => {
+    cy.get('.w-full').type("Berries")
+    cy.get('.text-white').click()
+    cy.get('.list-none li').should('have.length', 1)
+    cy.get('.list-none > :nth-child(1)').should('contain', 'Berries')
+    cy.get('#Complete').click({force: true})
```
Comment (tommy-anderson) [inline cypress/e2e/bugs.cy.ts:31]: is there a reason for `{force:true}` here? Most of the time it's an indication of an anti-pattern

File: cypress/e2e/create.cy.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+describe('Adding items test', () => {
+  beforeEach(() => {
+    cy.visit("/").get('.text-white').should('exist');
+  })
+
+  it('adds item to the list', () => {
+    cy.get('.w-full').type("Apples")
+    cy.get('.text-white').click()
+    cy.get('.list-none').should('have.length', 1)
+    cy.get('.list-none > :nth-child(1)').should('contain', 'Apples')
```
Comment (tommy-anderson) [inline cypress/e2e/create.cy.ts:12]: We asserted the name, but we did not assert the quantity and the unit here

File: cypress/e2e/create.cy.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    cy.get('.w-full').type("Apples")
+    cy.get('.w-20').type("99").type('{leftarrow}').type('{leftarrow}').type('{backspace}') //workaround for inserting the right quantity. 
+    cy.get('.text-white').click()
+    cy.get('.list-none').should('have.length', 1)
+    cy.get('.list-none > :nth-child(1)').should('contain', 'Apples')
+    cy.get('.list-none > :nth-child(1)').should('contain', '99')
+  })
+
+  it('adds item on Enter', () => {
+    cy.get('.w-full').type("Apples").type('{enter}')
```
Comment (tommy-anderson) [inline cypress/e2e/create.cy.ts:48]: `cy.type()` is [not safe](https://docs.cypress.io/api/commands/type#Yields) to chain further down the line, a better approach would be either
```
cy.get('.w-full').type('Apples{enter}')
```
or
```
cy.get('.w-full').type('Apples')
cy.get('.w-full').type('{enter}')
```

Comment (tommy-anderson) [inline cypress/e2e/create.cy.ts:48]: `cy.type()` is [not safe](https://docs.cypress.io/api/commands/type#Yields) to chain further down the line, a better approach would be either
```
cy.get('.w-full').type('Apples{enter}')
```
or
```
cy.get('.w-full').type('Apples')
cy.get('.w-full').type('{enter}')
```

File: cypress/e2e/edit.cy.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    cy.get('.list-none > :nth-child(1)').should('contain', 'Apples')
+  })
+
+  it('edits item from the list', () => {
+    cy.get('#Complete').click({force: true})
+    cy.get('.fixed > .bg-white').should('be.visible')
+    cy.get('.bg-white > .w-20').clear().type("2")
+    cy.get('.bg-white > .w-20').should('have.text', '2')
+    cy.get('.bg-white > select.shadow').select("cups")
+    cy.get('.bg-white > .w-full').clear().type('Cherries')
```
Comment (tommy-anderson) [inline cypress/e2e/edit.cy.ts:19]: `cy.clear()` is [not safe](https://docs.cypress.io/api/commands/clear#Yields) to chain further. A better approach would be 
```
cy.get('.bg-white > .w-full').clear()
cy.get('.bg-white > .w-full').type('Cherries')
```

Comment (VictorvManea) [reply cypress/e2e/edit.cy.ts:19]: Thanks for all the feedback! Have a lot of things to learn in TS/JS. This ground is a little fragile to me still, but with proper motivation and time, I guess everything is possible😄

**Pull request closed without merging**